coding-guidelines/coding-guidelines.md

```diff
@@ -0,0 +1,243 @@
+# Coding Guidelines
+
+## Line length
+One Line in JavaScript should have a maximal length of 80 characters.
+
+## Indentation and Line-Breaks
+Rules:
+- Indentation is done with 2 spaces in most cases.
+- Line breaks are only done if needed, try to use the 80 character limit.
+- Strings are splitted at the latest possible position.
+  - At the latest whitespace character.
+  - The second line is indented with 2 spaces.
+- Object Literals are kept on one line if possible.
+  - If the line gets to long put each attribute on a single line.
+  - Multi level object literals should be splitted in to multiple lines. Each level in the object should have 2 spaces as indentation.
+
+Examples:
+```JS
+switch (a) {
+  case "a":
+    break;
+  case "b":
+    break;
+}
+
+var a,
+  b,
+  c;
+
+function test() {
+  var a,
+    b,
+    c;
+}
+
+test(
+  "A Very long string which is longer then one line, which is 80 characters " +
+    "in this case. This demonstrates how the second line is indented",
+  "Another string parameter.",
+  {and: "a Literal Object", as: "a parameter", type: "short"},
+  {
+    yet: "another long parameter Object",
+    which: "needs to be split in multiple",
+    lines: 3
+  }
+);
+
+test({
+  a: {
+    b: 10
+  },
+  c: 20,
+  d: 30,
+  e: 40,
+  f: 50,
+  g: 60,
+  h: {
+    i: 70
+  }
+});
```

Comment:
I think it would be better to gather examples for things that can't easily be expressed with eslintrc rules, or that usually cause confusion in our PRs.